src/promptflow/promptflow/_sdk/_service/apis/flow.py

```diff
@@ -38,54 +42,103 @@
+            remove_dir = True
+        output_path = Path(output_path).resolve()
+        try:
+            if Configuration.get_instance().is_internal_features_enabled() and experiment:
+                result = get_client_from_request().flows.test(
+                    flow=flow,
+                    inputs=inputs,
+                    environment_variables=environment_variables,
+                    variant=variant,
+                    node=node,
+                    experiment=experiment,
+                    output_path=output_path,
+                    ux_call=True,
+                )
+                return_output = {}
+                for key in result:
+                    detail_path = output_path / key / "flow.detail.json"
+                    log_path = output_path / key / "flow.log"
+                    detail_content = json_load(detail_path)
+                    with open(log_path, 'r') as file:
+                        log_content = file.read()
+                    return_output[key] = {"detail": detail_content, "log": log_content}
+            else:
+                get_client_from_request().flows.test(
+                    flow=flow,
+                    inputs=inputs,
+                    environment_variables=environment_variables,
+                    variant=variant,
+                    node=node,
+                    allow_generator_output=False,
+                    stream_output=False,
+                    dump_test_result=True,
+                    output_path=output_path,
+                    ux_call=True,
+                )
+                if node:
+                    detail_path = output_path / f"flow-{node}.node.detail.json"
+                    log_path = output_path / f"{node}.node.log"
+                else:
+                    if variant:
+                        tuning_node, node_variant = parse_variant(variant)
+                        detail_path = output_path / f"flow-{tuning_node}-{node_variant}.detail.json"
+                    else:
+                        detail_path = output_path / "flow.detail.json"
+                    log_path = output_path / "flow.log"
+                detail_content = json_load(detail_path)
+                with open(log_path, 'r') as file:
+                    log_content = file.read()
+                return_output = {"flow": {"detail": detail_content, "log": log_content}}
+        finally:
+            if remove_dir:
+                shutil.rmtree(output_path)
+        return return_output
 
 
 @api.route("/get")
 class FlowGet(Resource):
     @api.response(code=200, description="Return flow yaml as json", model=dict_field)
     @api.doc(description="Return flow yaml as json")
     def get(self):
-        flow_path = api.payload["flow"]
+        args = flow_path_parser.parse_args()
+        flow_path = args.flow
+        flow_path = decrypt_flow_path(flow_path)
+        if not os.path.exists(flow_path):
+            return make_response("The flow doesn't exist", 404)
```

Comment:
Shall we raise some exception here? And there should be some common logic to handle execptions?
It feels strange by directly make_reponse here

Reply:
It seems have addded Exception formatted here. will raise error directly in api.
(https://github.com/microsoft/promptflow/blob/cc4a1c6fea4d6d59ff562b96968aab34dd9de4bd/src/promptflow/promptflow/_sdk/_service/utils/utils.py#L234C1-L234C26)